quickjs.c

```diff
@@ -2459,11 +2459,7 @@ static inline BOOL is_strict_mode(JSContext *ctx)
 
 static inline BOOL __JS_AtomIsConst(JSAtom v)
 {
-#ifdef DUMP_ATOM_LEAKS
-        return (int32_t)v <= 0;
-#else
-        return (int32_t)v < JS_ATOM_END;
-#endif
+    return (int32_t)v < JS_ATOM_END;
```

Comment:
I'm not 100% sure this is ok, but I don't see what the previous check tries to accomplish. AFAICT, the only case in which `(int32_t)v <= 0;` will be true is for tagged integer atoms. But then, in JS_FreeRuntime we only check for atoms >= JS_ATOM_END or if the refcount is not 1, so I think we're good?

Reply:
I suspect it's intended as an extra reference counting stress test.

An atom that's a keyword, like `Symbol("return")`, is constant and exempt from reference counting - but being exempt means refcount bugs won't surface.

(Aside: I don't like the int32_t cast, too clever and relies on implementation-defined behavior.)

Reply:
I suspected something similar. I don't have a strong opinion, do you have a preference here?

Reply:
If it doesn't complicate this PR, I'd keep it; otherwise, chuck it.

Reply:
It doesn't complicate things much, but I see little value in it plus there is the all-too-clever cast necessary. Look me a while to figure out what was going on because the code behaved differently in debug mode. All that to say I'm chucking it. We can always reconsider down the road if we get bitten by it :-P